services/clusters-keeper/src/simcore_service_clusters_keeper/modules/clusters_management_core.py

```diff
@@ -42,61 +74,89 @@ async def _find_terminateable_instances(
         app_settings.CLUSTERS_KEEPER_MAX_MISSED_HEARTBEATS_BEFORE_CLUSTER_TERMINATION
         * app_settings.SERVICE_TRACKING_HEARTBEAT
     )
+    startup_delay = (
+        app_settings.CLUSTERS_KEEPER_PRIMARY_EC2_INSTANCES.PRIMARY_EC2_INSTANCES_MAX_START_TIME
+    )
     for instance in instances:
-        last_heartbeat = _get_instance_last_heartbeat(instance)
-
-        elapsed_time_since_heartbeat = (
-            datetime.datetime.now(datetime.timezone.utc) - last_heartbeat
-        )
-        _logger.info(
-            "%s has still %ss before being terminateable",
-            f"{instance.id=}",
-            f"{(time_to_wait_before_termination - elapsed_time_since_heartbeat).total_seconds()}",
-        )
-        if elapsed_time_since_heartbeat >= time_to_wait_before_termination:
-            # let's terminate that one
-            terminateable_instances.append(instance)
+        if last_heartbeat := _get_instance_last_heartbeat(instance):
+            elapsed_time_since_heartbeat = arrow.utcnow().datetime - last_heartbeat
+            allowed_time_to_wait = time_to_wait_before_termination
```

Comment:
TIP: use defensive programming adding asserts on these times to avoid easy mistakes in the future e.g.

```python
assert elapsed_time_since_heartbeat > 0 # nosec
assert  elapsed_time_since_startup > 0 # nosec
```